client/src/partials/primary_cash/income/transfer/transfer.js

```diff
@@ -138,6 +141,23 @@ angular.module('bhima.controllers')
       .finally();
     };
 
+    function haltOnNoExchange () {
+      if (exchange.hasDailyRate()) { return; }
+
+      var instance = $modal.open({
+        templateUrl : 'partials/exchangeRateModal/exchangeRateModal.html',
+        backdrop    : 'static',
+        keyboard    : false,
+        controller  : 'exchangeRateModal'
+      });
+      
+      instance.result.then(function () {
+        $location.path('/exchange_rate');
```

Comment:
Since all of the call $location.path('/exchange_rate') on exit, I would put that in the exchangeRateModalController.

These changes look great!